contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,652 @@
+        expirationCallBounty = expirationCallBountyPercentage.mul(proposal.stakes[YES]).div(100);
+        require(stakingToken.transfer(msg.sender, expirationCallBounty));
+        emit ExpirationCallBounty(_proposalId,msg.sender,expirationCallBounty);
+    }
+
+    /**
+     * @dev redeem a reward for a successful stake, vote or proposing.
+     * The function use a beneficiary address as a parameter (and not msg.sender) to enable
+     * users to redeem on behalf of someone else.
+     * @param _proposalId the ID of the proposal
+     * @param _beneficiary - the beneficiary address
+     * @return rewards -
+     *           [0] stakerTokenReward
+     *           [1] voterReputationReward
+     *           [2] proposerReputationReward
+     */
+    function redeem(bytes32 _proposalId,address _beneficiary) public returns (uint[3] rewards) {
+        Proposal storage proposal = proposals[_proposalId];
+        require((proposal.state == ProposalState.Executed) || (proposal.state == ProposalState.Closed),"wrong proposal state");
+        Parameters memory params = parameters[proposal.paramsHash];
+        uint lostReputation;
+        if (proposal.winningVote == YES) {
+            lostReputation = proposal.preBoostedVotes[NO];
+        } else {
+            lostReputation = proposal.preBoostedVotes[YES];
+        }
+        lostReputation = (lostReputation * params.votersReputationLossRatio)/100;
+        //as staker
+        Staker storage staker = proposal.stakers[_beneficiary];
+        if (staker.amount > 0) {
+            if (proposal.state == ProposalState.Closed) {
+                //Stakes of a proposal that expires in Queue are sent back to stakers
+                rewards[0] = staker.amount;
+          }else if (staker.vote == proposal.winningVote) {
+                    uint totalWinningStakes = proposal.stakes[proposal.winningVote];
+                    uint totalStakes = proposal.stakes[YES]+proposal.stakes[NO];
+                    if (totalWinningStakes != 0) {
+                        if (staker.vote == YES) {
+                            uint _totalStakes = ((totalStakes*(100 - proposal.expirationCallBountyPercentage))/100) - proposal.daoBounty;
+                            rewards[0] = (staker.amount*_totalStakes)/totalWinningStakes;
+                        } else {
+                            rewards[0] = (staker.amount*totalStakes)/totalWinningStakes;
+                        }
+                 }
+          }
+            staker.amount = 0;
+        }
+        //as voter
+        Voter storage voter = proposal.voters[_beneficiary];
+        if ((voter.reputation != 0 ) && (voter.preBoosted)) {
+            uint preBoostedVotes = proposal.preBoostedVotes[YES] + proposal.preBoostedVotes[NO];
+            if (proposal.state == ProposalState.Closed) {
+              //give back reputation for the voter
+                rewards[1] = ((voter.reputation * params.votersReputationLossRatio)/100);
```

Comment:
not wanting to get into a meta-discussion, but are we sure we want to punish these voters here?

Reply:
we are not punishing them...just giving them back the voter the reputation it deposit earlier.

Reply:
you are givin back a percentage,  there is the `votersREputationLossRatio`.  It seems pointless to remove voters reputation just because they participated in a vote, in this case. (If this is something reasoned, just ignorme me here, but as I coudl not find it in the doc...)